erpnext/accounts/doctype/journal_entry/journal_entry.py

```diff
@@ -536,8 +605,7 @@ def get_against_jv(doctype, txt, searchfield, start, page_len, filters):
 	return frappe.db.sql("""select jv.name, jv.posting_date, jv.user_remark
 		from `tabJournal Entry` jv, `tabJournal Entry Account` jv_detail
 		where jv_detail.parent = jv.name and jv_detail.account = %s and ifnull(jv_detail.party, '') = %s
-		and (ifnull(jv_detail.against_invoice, '') = '' and ifnull(jv_detail.against_voucher, '') = ''
-		and ifnull(jv_detail.against_jv, '') = '' )
+		and (ifnull(jv_detail.reference_type, '') = ''
```

Comment:
I think this is fine the way it is. Allow for select against JV only if not referenced